### PR TITLE
Introduce support for NXP UCANS32K1SIC / S32K146

### PR DESCRIPTION
**Pull request by manuargue**
Introduce support for UCANS32K1SIC board, based on NXP S32K146 MCU. Support includes minimal set of drivers: clock control, pin control, MPU, GPIO and UART.
NXP S32K1 MCUs share a fair amount of similarities with Kinetis family, so some initialization code and configuration is based on existing code for Kinetis devices.

### CODE REVIEW

File: dts/arm/nxp/nxp_s32k146.dtsi [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	flash0: flash@0 {
+		compatible = "soc-nv-flash";
+		reg = <0 DT_SIZE_M(1)>;
+		erase-block-size = <DT_SIZE_K(4)>;
+		write-block-size = <8>;
+	};
+};
+
+&lpuart2 {
+	clocks = <&clock NXP_S32_LPUART2_CLK>;
```
[the remark this answers is not in the document]
Comment (manuargue) [reply dts/arm/nxp/nxp_s32k146.dtsi:46]: Not all parts have the same peripheral instance count, hence the clock definition wouldn't be available for those.

File: boards/arm/ucans32k1sic/ucans32k1sic.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+name: NXP UCANS32K1SIC
+vendor: nxp
+type: mcu
+arch: arm
+ram: 124
+flash: 1024
+toolchain:
+  - zephyr
+supported:
+  - mpu
```
[the remark this answers is not in the document]
Comment (manuargue) [reply boards/arm/ucans32k1sic/ucans32k1sic.yaml:14]: There are testes with these tags

File: dts/arm/nxp/nxp_s32k1xx.dtsi [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+			gpio-controller;
+			#gpio-cells = <2>;
+			nxp,kinetis-port = <&porte>;
+			status = "disabled";
+		};
+	};
+};
+
+&nvic {
+	arm,num-irq-priority-bits = <4>;
```
[the remark this answers is not in the document]
Comment (manuargue) [reply dts/arm/nxp/nxp_s32k1xx.dtsi:159]: moved to SoC-specific dtsi.

File: tests/drivers/uart/uart_async_api/boards/ucans32k1sic.overlay [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+dut: &lpuart2 {
```
[the remark this answers is not in the document]
Comment (manuargue) [reply tests/drivers/uart/uart_async_api/boards/ucans32k1sic.overlay:7]: right :sweat_smile: removed

File: dts/arm/nxp/nxp_s32k146.dtsi [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+/ {
+	cpus {
+		cpu@0 {
+			compatible = "arm,cortex-m4f";
+		};
+	};
+
+	soc {
+		/*
+		 * SRAM_L and SRAM_U ranges form a contiguous block but misaligned
```
[the remark this answers is not in the document]
Comment (manuargue) [reply dts/arm/nxp/nxp_s32k146.dtsi:19]: form a contiguous :)

**Review by bperseghetti — DISMISSED** [review]

File: soc/arm/nxp_s32/s32k1/soc.h [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+#include <S32K144W.h>
+#elif defined(CONFIG_SOC_S32K146)
+#include <S32K146.h>
+#elif defined(CONFIG_SOC_S32K148)
+#include <S32K148.h>
+#else
+#error "SoC not supported"
+#endif
+
+#if defined(CONFIG_CMSIS_RTOS_V2)
```
Comment (Dat-NguyenDuy) [inline soc/arm/nxp_s32/s32k1/soc.h:32]: Pls rebase main, [this](https://github.com/zephyrproject-rtos/zephyr/pull/65716/files#diff-e78dbc7cb3c75660e74759ac8c451f762446b3585f8759d3b4c76119368c6f7d) was merged

**Review by bperseghetti — DISMISSED** [review]

**Review by bperseghetti — APPROVED** [review]

**Review by Dat-NguyenDuy — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**